Persistent terminal

IceS2 opened the pull request:
## What does the PR do? (Required)

- [x] Added one Persisten Terminal
- [x] Fixed issue #11

Include information that will reviewers understand the changes better, now or in future.

## Checklist (Required)

- [x] I have tested the changes on my local machine
- [ ] I have added relevant documentation and tests for the changes
- [ ] I have followed the style guidelines of this project

## Evidence (Required)

https://github.com/user-attachments/assets/50aa93b6-2458-4d55-829c-180766802343

IceS2 committed:
Make Terminals Ephemeral

Since we are already opening a new terminal every time, I think it makes
sense to actually delete the buffer once it is hidden to avoid
accumulating hidden buffers in memory.

IceS2 committed:
Support Floating Terminal

Added support for simple floating terminals, without exposing
configurations. Breaking change on API just to rename Fterm and FTerm to
Tterm and TTerm, which I believe it makes more sense. Debatable!

IceS2 committed:
Revert "Make Terminals Ephemeral"

This reverts commit fda89e8dc299186434ef624f7dc8b527deee5f68.

IceS2 committed:
Add Support to having ONE persistent terminal

Although we are focusing on ephemeral terminals everywhere. I thnk it
could be useful to have one persistent terminal that would continue with
the context every time you opened it.

IceS2 committed:
Merge branch 'main' into persistent-terminal

IceS2 committed:
Fixed conflicts

2KAbhishek commented on plugin/termim.lua:
```diff
@@ -50,4 +52,20 @@ vim.api.nvim_create_user_command('VTerm', function(input)
+
+vim.api.nvim_create_user_command('TToggleTerm', function()
+    termim.toggle('tabnew')
+end, { nargs = '*' })
```
I don't want to introduce a new set of commands for this, lets instead use the existing

FTerm, TTerm, STerm and VTerm commands, we can also skip the persist flag and re-use keep_open instead

2KAbhishek reviewed (changes requested):
Left some comments for the approach, can you also make sure that the existing lint and test github actions steps are passing as well

I will be doing some tweaking once the PR is merged

Thanks for the PR

IceS2 commented on lua/termim/init.lua:
```diff
@@ -1,11 +1,24 @@
 
+termim.PersistentTerm = {
+    buf = nil,
+    win = nil,
```
My initial thought was to have one persistent terminal that I could open where I wanted, reusing the same one. If you believe we should have one persistent terminal per layout, I think this makes sense. I don't really have a strong opinion either way

IceS2 committed:
Implemented Feedback from PR

- We now have one persistent terminal per split type
- We now reuse the keep_open flag instead of creating yet a new one
- We now repurposed the current commands instead of creating a new set

IceS2 committed:
Fix tests by removing removed function
